src/containers/ExecSlave/Dockerfile

```diff
@@ -10,7 +10,15 @@ RUN mkdir -p $farmerHome/.ssh
 
 # Copy authorized_keys and set access for $user
 COPY [ "authorized_keys", "$farmerHome/.ssh/" ]
-RUN chmod 644 .ssh/authorized_keys && chown -R $user $farmerHome/.ssh && chgrp -R $user $farmerHome/.ssh
+
+# Some tests require that the slave can SSH to itself, setup keys
+RUN ssh-keygen -N "" -f $farmerHome/.ssh/id_rsa
+RUN cat $farmerHome/.ssh/id_rsa.pub >> $farmerHome/.ssh/authorized_keys
+
+# Set correct rights on .ssh
+RUN chmod 644 $farmerHome/.ssh/authorized_keys
+RUN chown -R $user $farmerHome/.ssh
+RUN chgrp -R $user $farmerHome/.ssh
```

Comment:
It is smart in Docker to combine RUN-commands on single line using && since that makes fewer layers (fewer hashes)